src/models/channel.js

```diff
@@ -0,0 +1,23 @@
+const mongoose = require("mongoose");
+const ObjectId = mongoose.Schema.Types.ObjectId;
+
+const ChannelSchema = mongoose.Schema(
+  {
+    name: {
+      type: String,
+      required: [true, "The channel name is required !"],
+      index: { unique: true },
+      minlength: [2, "The channel name must contain at least 2 characters !"]
+    },
+    ownerId: {
+      type: ObjectId,
+      required: true
+    },
+    tags: [String],
+    members: [ObjectId],
+    messages: [Object]
```

Comment:
This here needs a message schema instead of `Object`, probably just a text and creation date for the time being.
We would probably switch from `embedding` to `parent referencing` at one point because messages are bound to exceed the 16MB limit for a channel, but that can wait :+1:

Reply:
What about a message authorId ?

Reply:
One situation where that would, in my opinion, be a problem is, for example, on a username change (which may be a perfectly valid functionality in a project like ours). In that case, we need to update **all** the messages that have the old username which seems like a bad design to me.
I think it's better to have a `authorId` field instead.

Reply:
Well said